mypy/semanal.py

```diff
@@ -2748,6 +2748,13 @@ def analyze_name_lvalue(self,
         existing = names.get(name)
 
         outer = self.is_global_or_nonlocal(name)
+        if not outer and isinstance(self.type, TypeInfo) and self.type.is_enum:
+            # Special case: we need to be sure that `Enum` keys are unique.
+            if self.type.names.get(name) is not None:
+                self.fail('Attempted to reuse key "{}" in Enum definition "{}"'.format(
```

Comment:
Nit: In this context "reuse name" or "reuse member name" could be better than "reuse key", which implies a dictionary-like context (though key is acceptable since that mirrors the related exception message).

Reply:
I've copied the same error message we get from CPython:

```
» python ex.py
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/mypy/ex.py", line 3, in <module>
    class Foo(Enum):
  File "/Users/sobolev/Desktop/mypy/ex.py", line 5, in Foo
    B = A = 2
  File "/Users/sobolev/.pyenv/versions/3.9.1/lib/python3.9/enum.py", line 99, in __setitem__
    raise TypeError('Attempted to reuse key: %r' % key)
TypeError: Attempted to reuse key: 'A'
```

I'll add it! 👍